docs/src/user_docs/assembly/u32_operations.md

```diff
@@ -26,8 +26,8 @@ For instructions where one or more operands can be provided as immediate paramet
 | u32add <br> u32add.*b* | [b, a, ...] | [c, ...] | $c \leftarrow a + b$ <br> Fails if $max(a, b, c) \ge 2^{32}$ |
 | u32add.full    | [b, a, ...]    | [d, c, ...]   | $c \leftarrow (a + b) \mod 2^{32}$ <br> $d \leftarrow \begin{cases} 1, & \text{if}\ (a + b) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Fails if $max(a, b) \ge 2^{32}$ |
 | u32add.unsafe  | [b, a, ...]    | [d, c, ...]   | $c \leftarrow (a + b) \mod 2^{32}$, <br> $d \leftarrow \begin{cases} 1, & \text{if}\ (a + b) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Undefined if $max(a, b) \ge 2^{32}$ |
-| u32addc        | [b, a, c, ...] | [e, d, ...]   | $d \leftarrow (a + b + c) \mod 2^{32}$ <br> $e \leftarrow \begin{cases} 1, & \text{if}\ (a + b + c) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Fails if $max(a, b) \ge 2^{32}$ or $c > 1$ |
-| u32addc.unsafe | [b, a, c, ...] | [e, d, ...]   | $d \leftarrow (a + b + c) \mod 2^{32}$, <br> $e \leftarrow \begin{cases} 1, & \text{if}\ (a + b + c) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Undefined if $max(a, b) \ge 2^{32}$ <br> Fails if $c > 1$ |
+| u32add3        | [c, b, a, ...] | [e, d, ...]   | $d \leftarrow (a + b + c) \mod 2^{32}$, <br> $e \leftarrow \begin{cases} 2, & \text{if}\ (a + b + c) \ge 2^{64} \\ 1, & \text{if}\ (a + b + c) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Fails if $max(a, b, c) \ge 2^{32}$ |
+| u32add3.unsafe | [c, b, a, ...] | [e, d, ...]   | $d \leftarrow (a + b + c) \mod 2^{32}$, <br> $e \leftarrow \begin{cases} 2, & \text{if}\ (a + b + c) \ge 2^{64} \\ 1, & \text{if}\ (a + b + c) \ge 2^{32} \\ 0, & \text{otherwise}\ \end{cases}$ <br> Undefined if $max(a, b, c) \ge 2^{32}$ <br> |
```

Comment:
I don't think this is accurate: assuming `a`, `b`, and `c` are 32-bit values, `a + b + c` will never be greater than or equal to `2^64`. I'd use the same format here as we use  for `u32mul.unsafe`:
* `d <- (a + b + c) mod 2^32 `
* `e <- floor((a + b + c) / 2^32))`.

Reply:
Apologies! Yes, you are correct. It should be `2^33` instead of `2^64` but floor sounds better.

Please see [c01f9c9](https://github.com/maticnetwork/miden/pull/221/commits/c01f9c9f740150f19ac3b58150701c24a1941733)